src/include/ZividPython/Releasable.h

```diff
@@ -55,6 +56,13 @@
         return returnType{ impl().functionName(arg1Name) };                                                            \
     }
 
+#define ZIVID_PYTHON_FORWARD_1_ARGS_WRAP_RETURN_GIL_UNLOCKED(returnType, functionName, arg1Type, arg1Name)             \
+    auto functionName(arg1Type arg1Name)                                                                               \
+    {                                                                                                                  \
+        pybind11::gil_scoped_release gilLock;                                                                          \
```

Comment:
Idk much about dirty businesses with the GIL. However, since it is opt-in to disable it there must be some downsides. If not we could just add this line to all our macros instead of adding these special macros.

Can you extend the commit message to explain:

* What happens when we disable the GIL temporarily
* What the downsides are
* Why the downsides don't apply in this case
  * Or why the downsides are acceptable in this case

If we need to use these new macros with care also add a doc string per macro.

Reply:
@nedrebo Relevant excerpts from the [pybind11 website](https://pybind11.readthedocs.io/en/stable/advanced/misc.html):

> The Python C API dictates that the Global Interpreter Lock (GIL) must always be held by the current thread to safely access Python objects. As a result, when Python calls into C++ via pybind11 the GIL must be held, and pybind11 will never implicitly release the GIL.

This means pybind11 is being cautious to leave the GIL locked just in case we make any calls to any python functions/objects in our C++ code. But our C++ SDK does not access any Python objects, so it is perfectly safe to unlock the GIL for the duration of the function calls *to our SDK functions*. I don't see any down-sides to this at all.

> pybind11 will ensure that the GIL is held when it knows that it is calling Python code. For example, if a Python callback is passed to C++ code via std::function, when C++ code calls the function the built-in wrapper will acquire the GIL before calling the Python callback. Similarly, the PYBIND11_OVERRIDE family of macros will acquire the GIL before calling back into Python.
>
> When writing C++ code that is called from other C++ code, if that code accesses Python state, it must explicitly acquire and release the GIL.
>
> The classes gil_scoped_release and gil_scoped_acquire can be used to acquire and release the global interpreter lock in the body of a C++ function call. In this way, long-running C++ code can be parallelized using multiple Python threads, but great care must be taken when any gil_scoped_release appear: if there is any way that the C++ code can access Python objects, gil_scoped_acquire should be used to reacquire the GIL.

To make sure that if there is ever a python object in our *wrapper*, we make sure to unlock the GIL only for the duration of the function calls that go to the SDK. There reason it's done for this function, not elsewhere, is that it was needed here specifically. But in there is no reason to not do it for other blocking functions like `connect` and `disconnect`.

Reply:
Then I suggest KISS, just unlocking GIL in all the existing macros and postponing introducing a more complicated system until the day Zivid SDK actually calls into Python. That is unlikely to happen any time soon.

Thoughts @apartridge?

Reply:
Sounds good to me, I think we just have to be a bit dilligent with testing. We have quite good unit test coverage of the python wrapper, maybe also run the samples and see that it works fine?

Reply:
@nedrebo @apartridge I've extended it to all wrapper macros.